mezzanine/pages/admin.py

```diff
@@ -27,10 +27,11 @@ class PageAdminForm(DisplayableAdminForm):
     def clean_slug(self):
         """
         Save the old slug to be used later in PageAdmin.save_model()
-        to make the slug change propagate down the page tree.
+        to make the slug change propagate down the page tree, and clean
+        leading and trailing slashes which are added on elsewhere.
         """
         self.instance._old_slug = self.instance.slug
-        return self.cleaned_data['slug']
+        return clean_slashes(self.cleaned_data['slug'])
```

Comment:
Should only run `clean_slashes` here if the model isn't the Link model, something like `if not isinstance(self.instance, Link)`